### PR TITLE
feat(rnd): Simplify AI block

### PR DESCRIPTION
**Pull request by majdyz**
### Background

You basically need only a prompt field (and api_key, for now) for creating an AI block.

### Changes 🏗️

Make other fields on ai block optional.

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/blocks/ai.py [hunk 1]
```diff
@@ -14,17 +12,13 @@ class LlmModel(str, Enum):
...
-class LlmConfig(BaseModel):
-    model: LlmModel
-    api_key: str
-
-
 class LlmCallBlock(Block):
     class Input(BlockSchema):
-        config: LlmConfig
-        expected_format: dict[str, str]
+        api_key: str
```
Comment (majdyz) [inline rnd/autogpt_server/autogpt_server/blocks/ai.py:17]: I'll keep this for now until we have the creds setting.

File: rnd/autogpt_server/autogpt_server/blocks/ai.py [hunk 2]
```diff
@@ -37,55 +31,56 @@ def __init__(self):
...
-                    "model": "gpt-4-turbo",
-                    "api_key": "fake-api",
-                },
+                "model": "gpt-4-turbo",
+                "api_key": "fake-api",
                 "expected_format": {
                     "key1": "value1",
                     "key2": "value2",
                 },
-                "sys_prompt": "System prompt",
```
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/ai.py:48]: Don't we need to keep sys_prompt in the test_input?

**Comment by Torantulino** [comment]
So the frontend isn't playing nice with this.

Before running:
image
Notice that "Model" has no options.

After Running:

File: rnd/autogpt_server/autogpt_server/blocks/ai.py (hunk 2 above)
Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/blocks/ai.py:48]: Intentionally removed this to verify it still executes the block without the sys_prompt.

**Comment by majdyz** [comment]
I see, frontend doesn't support enum yet and it's not falling back to assume as a string? We can change it to str for now.


About the error:
I guess it's due to this 
https://github.com/Significant-Gravitas/AutoGPT/pull/7362/files#diff-8bc0ec9590a3208160360ef6789128eae490a405fa67349418d414dd0f2c4261R55

Instead of doing else {}
We should skip setting response_format param in the first place.

**Comment by Torantulino** [comment]
Model selection used to work just fine @majdyz, like this:

image

**Comment by Torantulino** [comment]
Okay, this is now functional except for "Model" not showing up as a dropdown anymore.
image

**Comment by majdyz** [comment]
Hmm probably a frontend bug? Providing default value should not change the schema, only some added metadata.

**Comment by aarushik93** [comment]
@majdyz can you show me what the API response looks like now please with the metadata?

**Comment by majdyz** [comment]
This is closed in favor of a larger change:
https://github.com/Significant-Gravitas/AutoGPT/pull/7373/files#diff-146658c7d8e69e8d71aa7eb64f0950731ad5a48ca3eb7d5d2e5225dabebe08f1R82-R87

**Pull request closed without merging**